codelab-master/apps/playground/src/app/quiz/containers/introduction/introduction.component.ts

```diff
@@ -0,0 +1,45 @@
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { QUIZ_DATA } from '@codelab-quiz/shared/quiz-data';
+import { Quiz } from '@codelab-quiz/shared/models/Quiz.model.ts';
+import { QuizService } from '@codelab-quiz/shared/services/quiz.service';
+
+
+@Component({
+  selector: 'codelab-quiz-intro',
+  templateUrl: './introduction.component.html',
+  styleUrls: ['./introduction.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class IntroductionComponent implements OnInit, OnDestroy {
+  quizData: Quiz[] = JSON.parse(JSON.stringify(QUIZ_DATA));
+  quizName = '';
+  unsubscribe$ = new Subject<void>();
+  imagePath = '../../../assets/images/';
+
+  constructor(private quizService: QuizService,
+              private activatedRoute: ActivatedRoute
+  ) { }
+
+  ngOnInit(): void {
+    this.activatedRoute.url
+      .pipe(takeUntil(this.unsubscribe$))
+        .subscribe(segments => {
```

Comment:
Can you use | async instead of subscribing?

Reply:
Ok I'll try to do this...

Reply:
Ok I fixed this issue.